src/Controllers/LinkFieldController.php

```diff
@@ -253,6 +263,69 @@ public function save(array $data, Form $form): HTTPResponse
         return $response;
     }
 
+    /**
+     * Update Link Sort fields based frontend drag and drop sort logic
+     * /admin/linkfield/sort
+     */
+    public function linkSort()
+    {
+        $request = $this->getRequest();
+        // Check security token
+        if (!SecurityToken::inst()->checkRequest($request)) {
+            $this->jsonError(400, _t(__CLASS__ . '.INVALID_TOKEN', 'Invalid CSRF token'));
+        }
+        $json = json_decode($request->getBody() ?? '', true);
+        $newLinkIDs = (array) $json['newLinkIDs'];
```

Comment:
```suggestion
        $json = json_decode($request->getBody() ?? '', false);
        // If someone's passing a JSON object or other non-array here, they're doing something wrong
        if (!is_array($newLinkIDs)) {
            $this->jsonError(400);
        }
        
        $newLinkIDs = $json->newLinkIDs;
```
See discussion in https://github.com/silverstripe/silverstripe-linkfield/pull/166#discussion_r1452757193

Reply:
Updated